backendAS/seedComments.js

```diff
@@ -1,36 +1,117 @@
+        const songlength = getRandomIntInclusive(120, 620);
+        this.songRowsGenerated++;
+        for (let i = 0; i < randomCommentNumber; i++) {
+          this.commentsGenerated++;
+          const songid = this.songRowsGenerated;
+          const commentimage = Math.floor(Math.random() * 200) + 1; // aws string link
+          const comment = faker.lorem.sentence();
+          const commenttime = getRandomIntInclusive(0, songlength); //presumably a number from song beginning aka 0 to song end
+          const username = faker.name.findName();
+          container += `${this.commentsGenerated},${songid},${commentimage},${comment},${commenttime},${username}\n`;
+        }
+        this.song(songlength);
       }
       this.push(container);
+      this._logPercentComplete();
+    }
+  }
+
+  async song(length) {
+    let songRow = '';
+    const songname = faker.lorem.sentence(2); // string
+    const artistname = faker.name.findName(); // String
+    const imgurl = Math.floor(Math.random() * 200) + 1; // string
+    const hashtag = faker.random.word();
+    const timeelapsed = getRandomIntInclusive(1543846091489, 1551730028863);
+    const decibel = getRandomIntInclusive(62, 80);
+    const songurl = 1; // aws string link
+
+    songRow += `${this.songRowsGenerated},${songname},${artistname},${imgurl},${hashtag},${timeelapsed},${length},${decibel},${songurl}\n`;
+    fs.appendFile('songs.csv', songRow, (err) => {
+      if (err) {
+        throw err;
+      }
+    })
+    //append to file songs.csv
+  }
+
+  _logPercentComplete() {
+    // const currPercent = ((this.songRowsGenerated / this.songsGoal) * 100).toFixed(2);
+    const currPercent = Math.floor((this.songRowsGenerated / this.songsGoal) * 100);
+    if (this.currPercent !== currPercent) {
+      this.currPercent = currPercent;
+      const hashes = this._getHashes(currPercent);
+      const spaces = this._getEmptySpaces(hashes);
+      const timestamp = moment().format('HH:mm:ss');
+      // const meta = `${timestamp} : ${this.filepath}`;
+      const progressBar = `Data generation: |${hashes}${spaces}| ${currPercent}% Complete`;
+
+      process.stdout.clearLine();
+      process.stdout.cursorTo(0);
+      if (currPercent === '100.00') {
+        process.stdout.write(`${timestamp} : ${progressBar}\n`);
+        console.log(`Success. Elapsed time since ${this.startTime.format('HH:mm:ss')}: ${moment().diff(this.startTime, 'seconds')} seconds\n`);
+      } else {
+        process.stdout.write(`${timestamp} : ${progressBar}`);
+      }
     }
   }
+
+  _getHashes(currPercent) {
+    return '#'.repeat(Math.floor(currPercent / (100 / this.progressBarLength)));
+  }
+
+  _getEmptySpaces(hashes) {
+    return ' '.repeat(this.progressBarLength - hashes.length);
+  }
 }
 
 
-var writestream = fs.createWriteStream('songs.csv');
-var readstream = new ReadSongs();
+var writestream = fs.createWriteStream('comments.csv');
+fs.unlink('songs.csv', (err) => {
```

Comment:
Consider refactor of fs use. If you want to overwrite any present file, you could simply write file once with no text. If you didn't want to overwrite every time, a method you could implement would be to append nothing to a file path (like songs-1.csv) and if no error is thrown try song-2, song-3, etc.